timemachine/fe/free_energy.py

```diff
@@ -95,34 +95,40 @@ class PairBarResult:
     """Results of BAR analysis on L-1 adjacent pairs of states given a sequence of L states."""
 
     initial_states: List[InitialState]  # length L
-    bar_results: List[BarResult]  # length L - 1
+
+    # length L - 1 pairs of u_kln_by_component, BAR result
+    # NOTE: BAR result may be None if estimator fails to converge
+    bar_results: List[Tuple[NDArray, Optional[BarResult]]]
```

Comment:
Note: this modification to the output schema is to allow for returning partial results (i.e. the $u_{kln}$) if the BAR computation fails to converge. But it will require downstream modifications if we keep it.

Reply:
What is preferable about changing the output schema over populating the BarResult objects with Nan values? Seems like these change require modifying a lot of code beyond just `esimate_free_energy_bar`